Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs

```diff
@@ -6,146 +6,153 @@
-                mail.CC.Add(ccRecipient);
-            }
-            //Add BCC recipients
-            foreach (var bccRecipient in bccRecipients)
-            {
-                mail.Bcc.Add(bccRecipient);
-            }
-
-            //Set message encoding
-            Encoding encoding = Encoding.GetEncoding(input.MessageEncoding);
-
-            mail.BodyEncoding = encoding;
-            mail.SubjectEncoding = encoding;
-
-            return mail;
-        }
-
-        /// <summary>
-        /// Gets all actual file names of attachments matching given file path
-        /// </summary>
-        private static ICollection<string> GetAttachmentFiles(string filePath)
-        {
-            string folder = Path.GetDirectoryName(filePath);
-            string fileMask = Path.GetFileName(filePath) != "" ? Path.GetFileName(filePath) : "*";
-
-            string[] filePaths = Directory.GetFiles(folder, fileMask);
-
-            return filePaths;
-        }
-    }
-}
+#pragma warning disable S101 // Types should be named in PascalCase
+	public static class SMTP
+#pragma warning restore S101 // Types should be named in PascalCase
+
+	{
+		/// <summary>
+		/// Sends email message with optional attachments.
+		/// [Documentation](https://github.com/FrendsPlatform/Frends.SMTP/tree/main/Frends.SMTP.SendEmail)
+		/// </summary>
+		/// <returns>
+		/// Object { bool EmailSent, string StatusString }
+		/// </returns>
+		public static Output SendEmail([PropertyTab] Input message, [PropertyTab] Attachment[] attachments, [PropertyTab] Options SMTPSettings, CancellationToken cancellationToken)
+		{
+			var output = new Output();
+
+			using (var client = InitializeSmtpClient(SMTPSettings))
+			{
+				using (var mail = InitializeMailMessage(message))
+				{
+					if (attachments != null)
+						foreach (var attachment in attachments)
+						{
+							if (attachment.AttachmentType == AttachmentType.FileAttachment)
+							{
+								ICollection<string> allAttachmentFilePaths = GetAttachmentFiles(attachment.FilePath);
+
+								if (attachment.ThrowExceptionIfAttachmentNotFound && allAttachmentFilePaths.Count == 0)
+									throw new FileNotFoundException(string.Format("The given filepath \"{0}\" had no matching files", attachment.FilePath), attachment.FilePath);
+
+								if (allAttachmentFilePaths.Count == 0 && !attachment.SendIfNoAttachmentsFound)
+								{
+									output.StatusString = string.Format("No attachments found matching path \"{0}\". No email sent.", attachment.FilePath);
+									output.EmailSent = false;
+									return output;
+								}
+
+								foreach (var fp in allAttachmentFilePaths)
+								{
+									mail.Attachments.Add(new System.Net.Mail.Attachment(fp));
+								}
+							}
+
+							if (attachment.AttachmentType == AttachmentType.AttachmentFromString
+								&& !string.IsNullOrEmpty(attachment.stringAttachment.FileContent))
+							{
+								mail.Attachments.Add(System.Net.Mail.Attachment.CreateAttachmentFromString
```

Comment:
Why do you need to specify the whole System.Net.Mail.Attachment.....?

Reply:
I need this because there is a class called Attachment and vs confuses the two Attachments, so I guess this is a better way than renaming the class to say mailAttachment?